### PR TITLE
Memory Game

### LINKED ISSUE #135: Memory game

### PR DESCRIPTION
**Pull request by sltsheryl**
Test on http://localhost:3000/test/memory
Fixes #135 

https://user-images.githubusercontent.com/96589109/209132799-4107e67f-6c7b-43c0-bfbf-f871dbe91821.mov

### CODE REVIEW

File: components/games/memory/CardFlipped.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (kevinchua6) [inline components/games/memory/CardFlipped.tsx:1]: Good usage of separation of concerns

File: components/games/memory/MemoryGame.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+const MemoryGame = () => {
+  const [randomCards, setRandomCards] = useState<CardObj[] | null>(null);
+  const [score, setScore] = useState<number>(0);
+  const [turns, setTurns] = useState<number>(0);
+  const [selectedCards, setSelectedCards] = useState<CardObj[]>([]);
+  const [isPlaying, setIsPlaying] = useState<boolean>(false);
+
+  const randomiseCards = () => {
```
Comment (kevinchua6) [inline components/games/memory/MemoryGame.tsx:20]: To make it more functional, probably can make `cards` a parameter for clarity

File: components/games/memory/Cards.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  name: string;
+  flipped: boolean;
+  found: boolean;
+}
+
+const Cards: Card[] = [
+  {
+    id: 1,
+    name: 'A',
+    flipped: false,
```
Comment (kevinchua6) [inline components/games/memory/Cards.tsx:12]: A small thing to take note during integration with BE (don't need to change now) - BE shouldn't need to pass `flipped` and `found` properties as they will always be false, can add it using the FE.

File: components/games/memory/Cards.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+interface Card {
+  id: number;
+  name: string;
+  flipped: boolean;
+  found: boolean;
+}
+
+const Cards: Card[] = [
```
Comment (kevinchua6) [inline components/games/memory/Cards.tsx:8]: Cards should be small case as its not a jsx component - `cards`

File: components/games/memory/Score.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+type PlayerInfoProps = {
+  turns: number;
+  score: number;
+};
+
+const Score = ({ turns, score }: PlayerInfoProps) => {
+  return (
+    <div>
+      <p>
+        {`Turns: `} {`${turns}`}
```
Comment (kevinchua6) [inline components/games/memory/Score.tsx:10]: Can combine the two -        {\`Turns: ${turns}\`}

File: package.json [hunk 6]
```diff
@@ -42,6 +42,7 @@
     "react-hook-form": "^7.34.2",
     "react-youtube": "^9.0.3",
     "recoil": "^0.7.5",
+    "styled-components": "^5.3.6",
```
Comment (kevinchua6) [inline package.json:45]: use tailwind and css!!!!! :(

**Review by kevinchua6 — CHANGES_REQUESTED** [review]
Really great job Sheryl! Thanks for the very quick PR. 

Small nit: also run `yarn` instead of `npm install`, ditch npm for this project as we should stick to one (I see that your package-lock.json is updated)

**Comment by sltsheryl** [comment]
I have made changes based on feedback already

**Review by kevinchua6 — APPROVED** [review]
LGTM, good job!

**Pull request merged**